applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1307,27 +1308,27 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateAndAddCompressibilityFlow(
 {
     KRATOS_TRY
 
-    this->CalculateCompressibilityFlow(rVariables.PPMatrix, rVariables.PVector, rVariables);
+    auto compressibility_flow = this->CalculateCompressibilityFlow(rVariables);
 
     // Distribute compressibility block vector into elemental vector
-    GeoElementUtilities::AssemblePBlockVector(rRightHandSideVector, rVariables.PVector);
+    GeoElementUtilities::AssemblePBlockVector(rRightHandSideVector, compressibility_flow);
 
     KRATOS_CATCH("")
 }
 
 template <unsigned int TDim, unsigned int TNumNodes>
-void UPwSmallStrainElement<TDim, TNumNodes>::CalculatePermeabilityFlow(
-    BoundedMatrix<double, TNumNodes, TNumNodes>& rPermeabilityMatrix,
-    array_1d<double, TNumNodes>&                 rPVector,
-    const ElementVariables&                      rVariables) const
+array_1d<double, TNumNodes> UPwSmallStrainElement<TDim, TNumNodes>::CalculatePermeabilityFlow(const ElementVariables& rVariables) const
 {
     KRATOS_TRY
 
-    rPermeabilityMatrix = GeoTransportEquationUtilities::CalculatePermeabilityMatrix<TDim, TNumNodes>(
-        rVariables.GradNpT, rVariables.DynamicViscosityInverse, rVariables.PermeabilityMatrix,
-        rVariables.RelativePermeability, rVariables.IntegrationCoefficient);
+    BoundedMatrix<double, TNumNodes, TNumNodes> permeability_matrix =
+        GeoTransportEquationUtilities::CalculatePermeabilityMatrix<TDim, TNumNodes>(
+            rVariables.GradNpT, rVariables.DynamicViscosityInverse, rVariables.PermeabilityMatrix,
+            rVariables.RelativePermeability, rVariables.IntegrationCoefficient);
 
-    noalias(rPVector) = -prod(rPermeabilityMatrix, rVariables.PressureVector);
+    array_1d<double, TNumNodes> result = -prod(permeability_matrix, rVariables.PressureVector);
```

Comment:
Also here, I think we might be able to combine this statement with the next `return` statement:
```suggestion
    return -prod(permeability_matrix, rVariables.PressureVector);
```

Or am I overlooking something?

Reply:
changed